utils/challenges/challenges.js

```diff
@@ -0,0 +1,36 @@
+import firebase from "../firebase/firebase";
+
+export async function setCurrentChallenge(teamId) {
+  // pick a random challenge
+  const challengeCountSnapshot = await firebase
+    .database()
+    .ref("/challenges/count")
+    .once("value");
+  const challengeCount = challengeCountSnapshot.val();
+  // challenge ids are indexed at 0
+  const challengeId = Math.floor(Math.random() * challengeCount);
+
+  const assignedTime = firebase.database.ServerValue.TIMESTAMP;
+  // time since the Unix epoch, in milliseconds) as determined by the Firebase servers
+  // see https://firebase.google.com/docs/reference/js/firebase.database.ServerValue
+
+  const newChallenge = {
+    challengeId,
+    teamId,
+    assignedTime
+  };
+
+  const assignedChallengeRef = await firebase
+    .database()
+    .ref(`/assignedChallenges/`)
+    .push(newChallenge);
+
+  const assignedChallengeId = assignedChallengeRef.key;
+
+  firebase
+    .database()
+    .ref(`/teams/${teamId}`)
+    .update({
```

Comment:
Because this function uses update, I think it would be a good idea to wrap it in a separate caller function that verifies that the `currentChallenge` field is empty before updating (or if an override flag is passed, updates regardless if empty). This we can we sure we don't overwrite active challenges accidentally.

something like
```
function updateCurrentChallenge(override=false){
        if(...checks if currentChallenge is empty... || override){
                ...do the update...
        }
        else {
                ...return an error message...        
        }
}
```

Reply:
Interesting but I feel is too much. I think the purpose of updateCurrentChallenge is to overwrite the previous challenge and I'm not sure why someone would call it in a non override way. This does bring up a good point though that archived challenges should probably be updated with the current challenge in the function though, so I'll add that.